### PR TITLE
[DDIDO-568]: Add SEARCH_URL environment variable support.

### PR DESCRIPTION
**Pull request by steveworley**
- Add environment variable support for SEARCH_URL

### CODE REVIEW

**Review by nicksantamaria — DISMISSED** [review]

File: bay/images/docker/settings.php [hunk 1]
```diff
@@ -256,6 +256,14 @@
 $config['clamav.settings']['mode_daemon_tcpip']['hostname'] = $clamav_host;
 $config['clamav.settings']['mode_daemon_tcpip']['port'] = $clamav_port;
 
+if ($lagoon_env_type != 'local' && getenv('SEARCH_AUTH_USER') && getenv('SEARCH_AUTH_PASSWORD')) {
+  $hash = getenv('SEARCH_HASH');
+  $search_url = getenv('SEARCH_URL') ?: 'nginx.sdp-elastic-nonprod.svc:8080';
+  $config['elasticsearch_connector.cluster.elasticsearch_bay']['url'] = sprintf('http://%s.%s', $hash, $search_url);
+  $config['elasticsearch_connector.cluster.elasticsearch_bay']['options']['username'] = getenv('SEARCH_AUTH_USER');
```
Comment (nicksantamaria) [inline bay/images/docker/settings.php]: This should be `SEARCH_AUTH_USERNAME`

File: bay/images/docker/settings.php [hunk 2]
```diff
@@ -256,6 +256,28 @@
 $config['clamav.settings']['mode_daemon_tcpip']['hostname'] = $clamav_host;
 $config['clamav.settings']['mode_daemon_tcpip']['port'] = $clamav_port;
 
+// Configure elasticsearch connections from environment variables.
+if ($lagoon_env_type != 'local') {
```
Comment (anthony-malkoun) [inline bay/images/docker/settings.php]: What about CI?

Comment (GROwen) [reply bay/images/docker/settings.php]: This is a good point and should be resolved by the same solution we come up for the FE (which is yet to be confirmed).

CI should behave the same as local for the moment (IIRC) so config would take care of it and there's no concern of the CMS writing to the Lagoon cluster. So can we just...

```suggestion
if ($lagoon_env_type !in_array(['local', 'ci'])) {
```

Comment (GROwen) [reply bay/images/docker/settings.php]: Confirmed that this change will leave CI behaviour untouched.

$lagoon_env_type is going to be populated in CI as `ci` through the CircleCI config of a project, which in turn is picked up by [Bay settings.php](https://github.com/dpc-sdp/bay/blob/7dd8751858c3b350e6baa613ba107da676d58a9f/bay/images/docker/settings.php#L11).

**Pull request merged**